teammapper-frontend/src/app/guards/toast.guard.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import {
+  CanActivate,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+  Router,
+} from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ToastGuard implements CanActivate {
+  constructor(private router: Router, private toastrService: ToastrService) {}
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Promise<boolean> | boolean {
```

Comment:
only use one type, e.g. `Promise<boolean>`. I don't think you need to change anything else, otherwise you can always a return a `Promise.resolve(true)`.